src/cpp/ext/otel/otel_server_call_tracer.cc

```diff
@@ -195,8 +196,14 @@ void OpenTelemetryServerCallTracer::RecordEnd(
 
 grpc_core::ServerCallTracer*
 OpenTelemetryServerCallTracerFactory::CreateNewServerCallTracer(
-    grpc_core::Arena* arena) {
-  return arena->ManagedNew<OpenTelemetryServerCallTracer>();
+    grpc_core::Arena* arena, const grpc_core::ChannelArgs& args) {
+  // Create the server call tracer only if there is no server selector
+  // registered or if the server selector returns true.
+  if (OTelPluginState().server_selector == nullptr ||
```

Comment:
Why are we looking at channel args on a per-call basis?

Reply:
Yeah, I was thinking about eventually having a Server-Channel scoped object tracer that would then create the per call `ServerCallTracer`..  Since we haven't agreed on the design there, I didn't want to jump the gun.

I guess, for now, I could create an API on the `ServerCallTracerFactory` for `IsServerChannelTraced()` and then save the bool on the Channel object? 

In the future, when we have more stuff on ServerChannel tracing, I could replace the bool with the `ServerChannelTracer`. What do you think?

Reply:
I don't think we should add anything to the `Channel` object.  That will increase per-channel memory on both client and server, even when OT is not in use.

I think we should just go ahead and create the server connection-scoped object.  None of this is public API yet, so we can change it later if we need to.

Reply:
> I don't think we should add anything to the Channel object. That will increase per-channel memory on both client and server, even when OT is not in use.

you're right. I got around this by just creating a `IsServerTraced()` method on the server call tracer factory, and saving it on the server. So, we only check it once per server.